src/bn_accounts.erl

```diff
@@ -24,75 +24,106 @@ handle_rpc(<<"account_get">>, {Param}) ->
                     {ok, OldLedger} ->
                         {V, OldLedger};
                     {error, height_too_old} ->
-                        {V,
-                            ?jsonrpc_error(
-                                {error, "height ~p is too old to get historic account data", [V]}
-                            )};
+                        case application:get_env(blockchain, store_historic_balances, false) of
+                            true ->
+                                {V, {error, height_too_old}};
+                            false ->
+                                {V, ?jsonrpc_error(
+                                    {error, "height ~p is too old to get historic account data", [V]}
+                                )}
+                        end;
                     {error, _} = Error ->
                         {V, ?jsonrpc_error(Error)}
                 end
         end,
     Address = ?jsonrpc_b58_to_bin(<<"address">>, Param),
-    GetBalance = fun() ->
-        case blockchain_ledger_v1:find_entry(Address, Ledger) of
-            {ok, Entry} ->
-                #{
-                    balance => blockchain_ledger_entry_v1:balance(Entry),
-                    nonce => blockchain_ledger_entry_v1:nonce(Entry),
-                    speculative_nonce => get_speculative_nonce(Address, balance, Ledger)
-                };
-            _ ->
-                #{
-                    balance => 0,
-                    nonce => 0,
-                    speculative_nonce => 0
-                }
-        end
-    end,
-    GetSecurities = fun() ->
-        case blockchain_ledger_v1:find_security_entry(Address, Ledger) of
-            {ok, Entry} ->
-                #{
-                    sec_balance => blockchain_ledger_security_entry_v1:balance(Entry),
-                    sec_nonce => blockchain_ledger_security_entry_v1:nonce(Entry),
-                    sec_speculative_nonce => get_speculative_nonce(
-                        Address,
-                        security,
-                        Ledger
-                    )
-                };
-            _ ->
-                #{
-                    sec_balance => 0,
-                    sec_nonce => 0,
-                    sec_speculative_nonce => 0
-                }
-        end
-    end,
-    GetDCs = fun() ->
-        case blockchain_ledger_v1:find_dc_entry(Address, Ledger) of
-            {ok, Entry} ->
+    case Ledger of
```

Comment:
I would try to refactor this giant match to a number of functions at this point. It's becoming unwieldy